src/utils/regex.cc

```diff
@@ -38,18 +38,21 @@ namespace modsecurity {
 namespace Utils {
 
 
-Regex::Regex(const std::string& pattern_)
+Regex::Regex(const std::string& pattern_, bool caseSensitive)
     : pattern(pattern_.empty() ? ".*" : pattern_) {
     const char *errptr = NULL;
     int erroffset;
+    int flags = (PCRE_DOTALL|PCRE_MULTILINE);
 
-    m_pc = pcre_compile(pattern.c_str(), PCRE_DOTALL|PCRE_MULTILINE,
+    if (caseSensitive == true) {
```

Comment:
The name of the variable is wrong: we add the CASELESS flag if caseSensitive is true. Either keep the name, pass false in the invocations of the constructor changed above and test for !caseSensitive here; or change the name to ignoreCase.

As written it will confuse countless readers :(

Reply:
Thank for your remark - you're right.

Because this PR contains so many unnecessary commit, and this name convention is really illogical, I created a new one - see #2297.

Now I close this PR. Thanks for all helps to everybody.